File: Projects/01/Ruby/rectangle_area.rb
# Project 1
# Emily Costa
# Feb 12 2020
###############################################################

# function rectangleArea returns a rectangle's area, given its height and width

def rectangleArea(h,w)
    h * w 
end

if __FILE__ == $0
   puts "\nTo compute the area of a rectangle,"
   print " enter its height: "
   height = gets.chomp.to_f
   print " enter its width: "
   width = gets.chomp.to_f
   print "\nThe circle's area is: "
   puts rectangleArea(height,width)
   print "\n"
end

File: Projects/01/Clojure/src/rectangle_area.clj

;;;; Project 1
;;;; Emily Costa
;;;; Feb 12 2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns rectangle_area) ; namespace function names the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn rectangleArea [height width]
  (* height width) )  ; returns area


;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main is where execution begins
;;; Input: the width and height of a rectangle.
;;; Output: the area of that rectangle.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (println "\nTo compute the area of a rectangle,")
  (print   " enter its height: ") (flush)
  (let [ height (read) ]
    (assert (>= height 0) "-main: height must be positive")  
    (print   " enter its width: ") (flush)
    (let [ width (read) ]

    (assert (>= width 0) "-main: width must be positive")

    (printf "\nThe area is %.2f \n\n" (float(rectangleArea height width)))
  )
))


